homeassistant/auth/providers/command_line.py

```diff
@@ -0,0 +1,160 @@
+"""Auth provider that validates credentials via an external command."""
+
+import typing as T
+
+import asyncio.subprocess
+import collections
+import logging
+import os
+
+import voluptuous as vol
+
+from homeassistant.exceptions import HomeAssistantError
+
+from . import AuthProvider, AUTH_PROVIDER_SCHEMA, AUTH_PROVIDERS, LoginFlow
+from ..models import Credentials, UserMeta
+
+
+CONF_COMMAND = "command"
+CONF_ARGS = "args"
+CONF_META = "meta"
+
+CONFIG_SCHEMA = AUTH_PROVIDER_SCHEMA.extend({
+    vol.Required(CONF_COMMAND): vol.All(
+        str,
+        os.path.normpath,
+        msg="must be an absolute path"
+    ),
+    vol.Optional(CONF_ARGS, default=None): vol.Any(vol.DefaultTo(list), [str]),
+    vol.Optional(CONF_META, default=False): bool,
+}, extra=vol.PREVENT_EXTRA)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class InvalidAuthError(HomeAssistantError):
+    """Raised when authentication with given credentials fails."""
+
+
+@AUTH_PROVIDERS.register("command_line")
+class CommandLineAuthProvider(AuthProvider):
+    """Auth provider validating credentials by calling a command."""
+
+    DEFAULT_TITLE = "Command Line Authentication"
+
+    # which keys to accept from a program's stdout
+    ALLOWED_META_KEYS = ("name",)
+
+    def __init__(self, *args: T.Any, **kwargs: T.Any) -> None:
```

Comment:
Would be a member of the type then, not of the instance... in terms of object isolation, that sound's not desirable, even it would make no difference as long as only ONE single instance of the auth provider exists at the same time - but I'm against it for said reasons.

Reply:
Aha, you are damn right.